skimage/exposure/_adapthist.py

```diff
@@ -167,24 +176,19 @@ def _clahe(image, kernel_size, clip_limit, nbins):
     hist = np.apply_along_axis(np.bincount, -1, hist_blocks, minlength=nbins)
     hist = np.apply_along_axis(clip_histogram, -1, hist, clip_limit=clim)
     hist = map_histogram(hist, 0, NR_OF_GRAY - 1, np.product(kernel_size))
-    hist = hist.reshape(hist_block_assembled_shape[:ndim] + (-1,))
-
-    # duplicate leading mappings in each dim
-    map_array = np.pad(hist,
-                       [[1, 1] for _ in range(ndim)] + [[0, 0]],
-                       mode='edge')
+    map_array = hist.reshape(hist_block_assembled_shape[:ndim] + (-1,))
 
     # Perform multilinear interpolation of graylevel mappings
-    # using the convention described here:
-    # https://en.wikipedia.org/w/index.php?title=Adaptive_histogram_
-    # equalization&oldid=936814673#Efficient_computation_by_interpolation
 
     # rearrange image into blocks for vectorized processing
-    ns_proc = [int(s / k) for s, k in zip(image.shape, kernel_size)]
+    ns_proc = [int(s / k) - 1 for s, k in zip(image.shape, kernel_size)]
```

Comment:
could you please explain this change?

Reply:
One block can be spared in the interpolation step, see illustration below.